syntax = "proto3";

import "google/protobuf/timestamp.proto";

service EDIService {
    rpc CalculateShipping(ShippingRequest) returns (stream ShippingResponse) {}
    rpc CreateOrder(CreateOrderRequest) returns (CreateOrderResponse) {}
    rpc CancelOrder(CancelOrderRequest) returns (CancelOrderResponse) {}
}

enum SERVICES {
    OTHER = 0;
    PAC = 1;
    SEDEX = 2;
}

message Item {
    int32 id = 1;
    string name = 2;
    double unit_price = 3;
    int32 quantity = 4;
}

message Order {
    repeated Item items = 1;
    double total_price = 2;
    google.protobuf.Timestamp date = 3;
    Customer customer = 4;
    Address shipping_address = 5;
}

message CreateOrderRequest {
    Order order = 1;
    SERVICES service = 2;
}

message Customer {
    string name = 1;
    string cpf = 2;
    string phonenumber = 3;
    string email = 4;
}

message Address {
    string street = 1;
    string district = 2;
    int32 number = 3;
    string city = 4;
    string cep = 5;
    string complement = 6;
}

message ShippingRequest {
    repeated Item items = 1;
    Address shipping_address = 2;
}

message ShippingResponse {
    int64 expected_days = 1;
    double shipping_cost = 2;
    string service = 3;
}

message CreateOrderResponse {
    bool sucess = 1;
    string tracking_code = 2;
}

message CancelOrderRequest {
    string order_id = 1;
}

message CancelOrderResponse {
    bool success = 1;
    string mensage = 2;
}

